tests/src/test/scala/com/nvidia/spark/rapids/GpuGenerateSuite.scala

```diff
@@ -0,0 +1,485 @@
+      inputSize +=
+        withResource(cvList.getChildColumnView(0)) {
+          _.getDeviceMemorySize
+        }
+      val batch = if (includeRepeatColumn) {
+        val dt: Array[DataType] = Seq(IntegerType, ArrayType(IntegerType)).toArray
+        val secondColumn = (0 until numRows).map { x =>
+          val i = Int.box(x)
+          if (allNulls || includeNulls && i % 2 == 0) {
+            null
+          } else {
+            i
+          }
+        }
+        withResource(ColumnVector.fromBoxedInts(secondColumn: _*)) { repeatCol =>
+          inputSize += listSize * repeatCol.getDeviceMemorySize
+          GpuColumnVector.from(new Table(repeatCol, cvList), dt)
+        }
+      } else {
+        val dt: Array[DataType] = Seq(ArrayType(IntegerType)).toArray
+        GpuColumnVector.from(new Table(cvList), dt)
+      }
+      (batch, inputSize)
+    }
+  }
+
+  def checkSplits(splits: Array[Int], batch: ColumnarBatch): Unit = {
+    withResource(GpuColumnVector.from(batch)) { tbl =>
+      var totalRows = 0L
+      // because concatenate does not work with 1 Table and I can't incRefCount a Table
+      // this is used to close the concatenated table, which would otherwise be leaked.
+      withResource(new ArrayBuffer[Table]) { tableToClose =>
+        withResource(tbl.contiguousSplit(splits: _*)) { splitted =>
+          splitted.foreach { ct =>
+            totalRows += ct.getRowCount
+          }
+          // `getTable` causes Table to be owned by the `ContiguousTable` class
+          // so they get closed when the `ContiguousTable`s get closed.
+          val concatted = if (splitted.length == 1) {
+            splitted(0).getTable
+          } else {
+            val tbl = Table.concatenate(splitted.map(_.getTable): _*)
+            tableToClose += tbl
+            tbl
+          }
+          // Compare row by row the input vs the concatenated splits
+          withResource(GpuColumnVector.from(batch)) { inputTbl =>
+            assertResult(concatted.getRowCount)(batch.numRows())
+            (0 until batch.numCols()).foreach { c =>
+              withResource(concatted.getColumn(c).copyToHost()) { hostConcatCol =>
+                withResource(inputTbl.getColumn(c).copyToHost()) { hostInputCol =>
+                  (0 until batch.numRows()).foreach { r =>
+                    if (hostInputCol.isNull(r)) {
+                      assertResult(true)(hostConcatCol.isNull(r))
+                    } else {
+                      if (hostInputCol.getType == DType.LIST) {
+                        // exploding column
+                        compare(hostInputCol.getList(r), hostConcatCol.getList(r))
+                      } else {
+                        compare(hostInputCol.getInt(r), hostConcatCol.getInt(r))
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  test("all null inputs") {
```

Comment:
From my recent experience with CI test, I would raise more awareness about the memory leak issue with new functionalities. So please check for the issue with these tests running locally to make sure we won't have it.

Reply:
@ttnghia thanks for the reminder. You helped catch a test leak! See: https://github.com/NVIDIA/spark-rapids/pull/6631/commits/c2c1676e7f2eb36397a047871240a311ed061dc0